src/settings.js

```diff
@@ -0,0 +1,567 @@
+"use strict";
+(function () {
+    var fluid = window.fluid,
+        gpii = fluid.registerNamespace("gpii"),
+        ipcRenderer = require("electron").ipcRenderer;
+
+    fluid.registerNamespace("gpii.app.settings");
+
+    ipcRenderer.on("message", function (event, message) {
+        console.log("Browser window received message:", message);
+    });
+
+    gpii.app.settings.getIcon = function (preferenceSets, activePreferenceSet) {
+        if (!gpii.app.settings.hasPreferenceSets(preferenceSets)) {
+            return;
+        }
+
+        for (var i = 0; i < preferenceSets.names.length; i++) {
+            if (preferenceSets.names[i] === activePreferenceSet) {
+                return preferenceSets.icons[i];
+            }
+        }
+    };
+
+    gpii.app.settings.addCommunicationChannel = function (that) {
+        that.updatePreferenceSets({
+            names: ["GPII Default", "Subway", "Noisy"],
```

Comment:
Even though this is just scaffolding material, it may as well still go into component options rather than inline values